### DISCUSSION TITLE
no graph showing

### DISCUSSION THREAD

**Discussion by TP1961** [artisan-roaster-scope/artisan#836] [Q&A]
Hi need a little help trying to hook Artisan up to my SF1 cant get a graph to show? When roasting if I hit dry phase button, and 1st crack button, it'll show my development time. I configured it for the SF roaster, not sure what else to do. I cant get a graph to show

**Comment by roasterdave** [comment]
Artisan works great with San Franciscans so there is something in your setup that is not correct. You will need to provide some more information for me to help.

What version of Artisan are you running (Help>> About). What OS are you using?

There are no curves at all shown in your profile graph? A screenshot during a roast will be helpful here. Also, please attach your current Artisan settings file (as exported via menu Help >> Save Settings as *.aset) file. Note that you need either to add a .txt extension or zip Artisan files before uploading. Otherwise you will receive a "Not a supported file type" error on uploading. Files need to be attached on the website, attaching them to email replies will not work.

Hitting the DRY and FCs buttons set the time of these events. Development time is calculated from the FCs time and the total time so this behavior is expected.

**Reply by TP1961** [reply]
Hi Dave what is a txt extension

**Reply by roasterdave** [reply]
By default Artisan will save the settings file to "artisan-settings.alog".   Rename the file to "artisan-settings.txt".  Then come to this Github website and drag-and-drop the file on a comment field to upload it.

**Reply by TP1961** [reply]
sorry, sent file to your email. can't seem to send it on GitHub.

**Reply by TP1961** [reply]
![Tim Screenshot 3-10-2022](https://user-images.githubusercontent.com/101024081/157664593-568a0d01-0005-4568-ae4b-47da3682d4e1.png)
![Screenshot (15)](https://user-images.githubusercontent.com/101024081/157664607-c81adf60-6776-4531-a6b1-5ef09c1792f8.png)
![Screenshot (14)](https://user-images.githubusercontent.com/101024081/157664613-60f4f176-c326-4eb1-89aa-9b8c93f589ff.png)

**Reply by TP1961** [reply]
[blackbird1(.zip](https://github.com/artisan-roaster-scope/artisan/files/8223501/blackbird1.zip)

**Reply by roasterdave** [reply]
Artisan is not receiving temperatures from the SF1.  Notice the message in the upper left that says "Modbus Communication Error".  That means there is no communication from the roaster to Artisan.  

Did you install the Windows virtual comm driver that came with the roaster?  This driver is needed for Windows to communicate with the RS485-USB adapter that came with your roaster.  The driver is also available [here](https://ftdichip.com/drivers/vcp-drivers/).  

After the driver is installed you will need to set the proper Modbus serial port in Artisan's settings.  Be sure you are connected to your roaster and that the roaster is turned on and showing temperatures on the roaster's display.  You do not need to turn on the the gas.  From the menu in Artisan choose **Port** from the **Config** menu.  Then click on the **Modbus** tab.  See the screenshot below.  Your Windows machine assigns the port numbers so I can't tell you which is the correct one to use.  You will need to try each, one by one, as follows.  Select a port from the pull down highlighted in the screenshot.  Click OK then hit Artisan's ON button.  If you see ET and BT temperatures displayed in the LCDs  you will know you have selected the correct port.  If you see 'uu' then go back and try the next port.  

Let me know what happens.  If none of the ports work you may need to contact San Franciscan for additional help since the problem is outside of Artisan. 

![Screenshot - 3_10_2022 , 7_20_48 AM](https://user-images.githubusercontent.com/30009312/157700658-0a8a45a1-6190-44f1-b7d4-1575148a7fd4.png)

**Reply by poundy** [reply]
Another way to figure out exactly what COM port you need to select is to open Device Manager and look at the "Ports (COM & LPT)" section, as you plug the cable into the roaster.  In Device Manager, it hides the "Ports (COM & LPT)" section unless there's a device in that, so when you first open it don't be surprised if that doesn't appear. If it is listed, expand it and see what ports you have listed there - the roaster will NOT be visible at this point, so you're trying to see what new port appears in the next step.  
Connect the roaster via the cable, and then you'll either see this section appear or the new port will be listed.  If you need to expand that section, do so; take note of the new COM port that became visible, and select that in the Artisan configuration

**Comment by TP1961** [comment]
Hi Dave, I have Windows 10 on HP laptop. Using Artisan 2.4.6 from Sweet Maria's. No graphs showing on profile. I'll have to get a screenshot when I am roasting. I'm not sure what Artisan setting file I have, could'nt find it in the help section of the menu. Part of the problem is I know very little about computers. Sent from my  U.S.Cellular© Smartphone
-------- Original message --------From: Dave Baxter ***@***.***> Date: 3/5/22  5:54 PM  (GMT-06:00) To: artisan-roaster-scope/artisan ***@***.***> Cc: TP1961 ***@***.***>, Author ***@***.***> Subject: Re: [artisan-roaster-scope/artisan] no graph showing (Discussion #836) 
Artisan works great with San Franciscans so there is something in your setup that is not correct. You will need to provide some more information for me to help.
What version of Artisan are you running (Help>> About). What OS are you using?
There are no curves at all shown in your profile graph? A screenshot during a roast will be helpful here. Also, please attach your current Artisan settings file (as exported via menu Help >> Save Settings as *.aset) file. Note that you need either to add a .txt extension or zip Artisan files before uploading. Otherwise you will receive a "Not a supported file type" error on uploading. Files need to be attached on the website, attaching them to email replies will not work.
Hitting the DRY and FCs buttons set the time of these events. Development time is calculated from the FCs time and the total time so this behavior is expected.

**Reply by roasterdave** [reply]
To save your settings go to the Help menu and choose Save Settings.  

![Screenshot - 3_7_2022 , 5_14_24 PM](https://user-images.githubusercontent.com/30009312/157146265-09d14825-dde2-479a-b68e-95cf107bc97d.png)

**Comment by TP1961** [comment]
OK will do that tonight I should be good then?  thanks Dave.Tim
    On Monday, March 7, 2022, 07:16:05 PM CST, Dave Baxter ***@***.***> wrote:  
 
 


To save your settings go to the Help menu and choose Save Settings.



—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by poundy** [reply]
no, once you save the settings, then come back here (github page) and attach the things asked for to this thread, and people can look at the details and start to figure out what you have not done. 

While the email response seems like it will work, it doesn't handle replies and adds extra text so if you could, keep coming back here to github and reply, don't use email if you can avoid it (you have to come here to send the attachments we need anyway)

**Reply by MAKOMO** [reply]
to attach the artisan-settings.aset file you need to rename it to artisan-settings.txt and drag-and-drop it here on Github on a comment field for upload.

**Reply by TP1961** [reply]
so you need to know the file name?

**Reply by TP1961** [reply]
how would I do that

**Comment by TP1961** [comment]
Ok renamed fileSent from my  U.S.Cellular© Smartphone
-------- Original message --------From: Dave Baxter ***@***.***> Date: 3/7/22  7:16 PM  (GMT-06:00) To: artisan-roaster-scope/artisan ***@***.***> Cc: TP1961 ***@***.***>, Author ***@***.***> Subject: Re: [artisan-roaster-scope/artisan] no graph showing (Discussion #836) 
To save your settings go to the Help menu and choose Save Settings.

**Reply by poundy** [reply]
Clearly you didn't read my plea - stop using email !

And because you replied in email, even if you think you attached a file to the email reply, it never made it here. Please, stop replying in email and just visit the "view it on github" link in the email. 

Just repeating what we are asking for: 
* Please save your settings file, rename it with a ".txt" extension, and attach to a reply here.
* Please take a screenshot of what you mean by "can't get a graph to show". The whole screen please, nothing missing.  Drop that screenshot on the reply as well.

**Reply by TP1961** [reply]
sorry about that. will try to get a screenshot tonight and file name.

**Comment by TP1961** [comment]
Hope this is what you mean Dave. I’ll get a screenshot when I do my roast in a bit Sent from Mail for Windows From: Dave BaxterSent: Tuesday, March 8, 2022 11:49 PMTo: artisan-roaster-scope/artisanCc: TP1961; AuthorSubject: Re: [artisan-roaster-scope/artisan] no graph showing (Discussion #836) By default Artisan will save the settings file to "artisan-settings.alog". Rename the file to "artisan-settings.txt". Then come to this Github website and drag-and-drop the file on a comment field to upload it.—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android. You are receiving this because you authored the thread.Message ID: ***@***.***>

**Reply by poundy** [reply]
PLEASE STOP replying in email ! 

None of what you are doing here is helping. Come to the github link at the bottom of the email and attach images and files here directly.

**Comment by MAKOMO** [comment]
Just add the extension .txt to the filename and drag and drop the file on Github to a Comment/Write/Answer field

**Reply by TP1961** [reply]
let me know if you can open thanks

**Comment by TP1961** [comment]
Hi guys have not been back on here work and family keep me away.  I saw there was a list of things I was to try would it be possible to put then back? up sorry about that.Tim

**Reply by roasterdave** [reply]
Tim, all of the previous posts are still available on this discussion page.  Visit with a browser to see all the history.   If you see "view more" you should click on it because sometimes a reply may be hidden.  

Here is the link to the page https://github.com/artisan-roaster-scope/artisan/discussions/836.   You can also click the "view it on GitHub" link at the bottom of the emails you receive.

**Reply by TP1961** [reply]
Dave I clicked on number and it took me to Discussion page were do I get to my post/?

**Reply by poundy** [reply]
Sigh. You are IN your post !! If you scroll up you will see the full history of this discussion

**Reply by TP1961** [reply]
Hi guys, I think we got it! I want to thank you for all of your help, and patience!

![Screenshot (18)](https://user-images.githubusercontent.com/101024081/160302690-c2f129a6-4cb1-492d-b210-2ac48309a744.png)

**Reply by roasterdave** [reply]
Excellent! Thanks for letting us know.

**Reply by TP1961** [reply]
now I need to know what I am looking at.Was watching the screen more than the roaster so how bad dose this look for a roast?

**Reply by poundy** [reply]
Well, as much as graphs are nice, and this is a discussion about Artisan, we can't tell you how your coffee tasted, because that is *all*that*matters*
